File: Data Structures & Algorithms/remove-element/submission-0.go
func removeElement(nums []int, val int) int {
    count := 0
	for _, num := range nums { if num == val { count++ }}
	n := len(nums)
	for i := 0; i < n; i++ {
		for i < n && nums[i] != val {
			i++
		}

		j := i + 1
		for j < n && nums[j] == val {
			j++
		}

		if i < n && j < n {
			nums[i], nums[j] = nums[j], nums[i]
		}
	}

	return n - count
}

File: Data Structures & Algorithms/sort-colors/submission-0.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        sortByNumber(nums, 1, sortByNumber(nums, 0, 0) );
    }

    int sortByNumber(vector<int>& nums, int num, int index) {
        for ( int i = 0; i < nums.size(); i++ ) {
            if ( nums[i] == num ){
                swap(nums[index], nums[i]);
                index++;
            }
        }

        return index;
    }
};